server/server.go

```diff
@@ -510,17 +510,26 @@ func (s *Server) startMonitoring(secure bool) error {
 		hp           string
 		err          error
 		httpListener net.Listener
+		port   int
 	)
 
 	if secure {
-		hp = net.JoinHostPort(opts.HTTPHost, strconv.Itoa(opts.HTTPSPort))
+		port = opts.HTTPSPort
+		if port == -1 {
+			port = 0
+		}
```

Comment:
The cluster port should follow the same strat. The next bit that is interesting is introspecting this output from the launch process (maybe with the option to store this is a file) - In some cases the launch will happen by a system service and capturing this in a way that is useful (even on actually specified ports) would be very useful.